### PR TITLE
[develop] Add Hera GNU modulefile and fundamental test suite and update Hera Intel modulefile.

### LINKED ISSUE #476: Adding a GNU modulefile for Hera
## Description
There is currently no GNU modulefile for Hera (i.e. `build_hera_gnu.lua`).  For developers who don't have access to Cheyenne, it is impossible to test GNU functionality.  However, there are GNU and mpich available on Hera, so adding this capability to Hera would allow those developers to test GNU issues on Hera.

## Solution
Add `build_hera_gnu.lua` to the available modulefiles.

## Requirements**
Everything that is required to build the SRW App using the GNU compiler are available on Hera.

## Acceptance Criteria (Definition of Done)
The SRW App builds successfully on Hera using GNU compiler `9.2.0`.

### PR DESCRIPTION
**Pull request by MichaelLueken**
## DESCRIPTION OF CHANGES: 
Currently, the only machine that has a functioning GNU modulefile is Cheyenne.  It has been noted that several developers don't have access to Cheyenne, so if there are issues with GNU compilers, it is difficult, if not impossible, to debug.  All of the necessary libraries are available on Hera to compile the SRW App using `GNU 9.2.0` with `MPICH 3.3.2`.  A new `build_hera_gnu.lua` modulefile has been created so that developers who can't access Cheyenne are still able to test their changes on another machine using the GNU compiler.

The work that @danielabdi-noaa has done in PR #479 has also been merged into this PR so that fundamental tests are ready for the GNU compiler on Hera.

Updated `modulefiles/build_hera_intel.lua` to point to the new EPIC-maintained HPC-stack locations.  Fundamental tests have been run and all successfully passed on Hera.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 
- [X] hera.gnu
The SRW App successfully compiled on Hera using GNU 9.2.0.  Fundamental tests have successfully completed.
- [X] hera.intel
The SRW App successfully compiled using the new EPIC-maintained HPC-stack location for Intel.  Fundamental tests have successfully completed.
- [ ] fundamental test suite

## ISSUE: 
Completes #476 

## CHECKLIST
- [X] My code follows the style guidelines in the Contributor's Guide
- [X] I have performed a self-review of my own code using the Code Reviewer's Guide
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [X] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## CONTRIBUTORS
@danielabdi-noaa - Created a fundamental test suite for Hera with GNU compiler.
@natalie-perlin - Setting up the EPIC-maintained HPC-stack locations on Hera.

### CODE REVIEW

**Comment by venitahagerty** [comment]
Machine: hera
Compiler: gnu
Job: build
Repo location: /scratch1/BMC/zrtrr/rrfs_ci/autoci/pr/1122018556/20221114213516/ufs-srweather-app
Build was Successful
If test failed, please make changes and add the following label back:
ci-hera-gnu-build

File: modulefiles/build_hera_gnu.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+load(pathJoin("hpc-mpich", os.getenv("hpc-mpich_ver") or "3.3.2"))
+
+load("srw_common")
+
+load(pathJoin("nccmp", os.getenv("nccmp_ver") or "1.8.7"))
+load(pathJoin("nco", os.getenv("nco_ver") or "4.9.3"))
+
+setenv("CMAKE_C_COMPILER","mpiicc")
+setenv("CMAKE_CXX_COMPILER","mpiicpc")
+setenv("CMAKE_Fortran_COMPILER","mpiifort")
```
Comment (danielabdi-noaa) [inline modulefiles/build_hera_gnu.lua]: These should be the GNU couterparts i think
```Lua
setenv("CMAKE_C_COMPILER","mpicc")
setenv("CMAKE_CXX_COMPILER","mpicxx")
setenv("CMAKE_Fortran_COMPILER","mpif90")
```

**Comment by MichaelLueken** [comment]
@danielabdi-noaa Nice catch!  I have corrected this entries.

**Comment by venitahagerty** [comment]
Machine: hera
Compiler: gnu
Job: WE
Repo location: /scratch1/BMC/zrtrr/rrfs_ci/autoci/pr/1122018556/20221114223509/ufs-srweather-app
Build was Successful
Rocoto jobs started
Long term tracking will be done on 10 experiments
If test failed, please make changes and add the following label back:
ci-hera-gnu-WE
Experiment Succeeded on hera: community_ensemble_2mems_stoch
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2
Experiment Succeeded on hera: grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR
Experiment Succeeded on hera: pregen_grid_orog_sfc_climo
Experiment Succeeded on hera: grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
Experiment Failed on hera: MET_ensemble_verification
2022-11-14 23:52:08 +0000 :: hfe10 :: Task run_ensgridvx, jobid=37805369, in state DEAD (FAILED), ran for 33.0 seconds, exit status=256, try=1 (of 1)
All experiments completed

**Review by danielabdi-noaa — APPROVED** [review]

File: modulefiles/build_hera_gnu.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+the NOAA RDHPC machine Hera using GNU 9.2.0
+]])
+
+whatis([===[Loads libraries needed for building the UFS SRW App on Hera using GNU 9.2.0 ]===])
+
+prepend_path("MODULEPATH","/contrib/sutils/modulefiles")
+load("sutils")
+
+load(pathJoin("cmake", os.getenv("cmake_ver") or "3.20.1"))
+
```
Comment (natalie-perlin) [inline modulefiles/build_hera_gnu.lua:12]: New hpc-stack location in EPIC-managed space. As discussed in [SRW-PR-419](https://github.com/ufs-community/ufs-srweather-app/pull/419), [SRW-Issue-427](https://github.com/ufs-community/ufs-srweather-app/issues/427), [UFS-WM-Issue-1465](https://github.com/ufs-community/ufs-weather-model/issues/1465), [UFS-WM-PR-1468](https://github.com/ufs-community/ufs-weather-model/pull/1468).
This stack in this location on Hera has been independently tested with the UFS-WM by @jkbk2004, and not changing the baseline. The following lines 13-16 would be changed to the following:

```
gnu_ver=os.getenv("gnu_ver") or "9.2.0"
load(pathJoin("gnu", gnu_ver))

prepend_path("MODULEPATH", "/scratch1/NCEPDEV/nems/role.epic/hpc-stack/libs/gnu-9.2/modulefiles/stack")

load(pathJoin("hpc", os.getenv("hpc_ver") or "1.2.0"))
load(pathJoin("hpc-gnu", os.getenv("hpc-gnu_ver") or "9.2"))
```

Note the **("hpc-gnu_ver") or "9.2"))**

File: modulefiles/build_hera_gnu.lua [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+prepend_path("MODULEPATH", "/scratch2/NCEPDEV/nwprod/hpc-stack/libs/hpc-stack/modulefiles/stack")
+
+load(pathJoin("hpc", os.getenv("hpc_ver") or "1.2.0"))
+load(pathJoin("hpc-gnu", os.getenv("hpc-gnu_ver") or "9.2.0"))
+load(pathJoin("hpc-mpich", os.getenv("hpc-mpich_ver") or "3.3.2"))
+
+load("srw_common")
+
+load(pathJoin("nccmp", os.getenv("nccmp_ver") or "1.8.7"))
```
Comment (natalie-perlin) [inline modulefiles/build_hera_gnu.lua]: The version built with the new stack: 1.8.9 (or 1.9.1)
`load(pathJoin("nccmp", os.getenv("nccmp_ver") or "1.8.9"))`

**Review by natalie-perlin — CHANGES_REQUESTED** [review]
New hpc-stack location in EPIC-managed space. As discussed in https://github.com/ufs-community/ufs-srweather-app/pull/419, https://github.com/ufs-community/ufs-srweather-app/issues/427, https://github.com/ufs-community/ufs-weather-model/issues/1465, https://github.com/ufs-community/ufs-weather-model/pull/1468.
This stack in this location on Hera has been independently tested with the UFS-WM by @jkbk2004, and not changing the baseline.

**Comment by natalie-perlin** [comment]
A test with EPIC-managed stack has successfully completed on Hera. Test directory: /scratch1/NCEPDEV/nems/role.epic/sandbox/SRW/expt_dirs/test_gnu/grid_RRFS_CONUScompact_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16

**Comment by MichaelLueken** [comment]
@natalie-perlin Thank you very much for your review!  I have made the suggested changes so that the Hera GNU modulefile will use the new EPIC maintained HPC-stack location.

**Comment by venitahagerty** [comment]
Machine: hera
Compiler: gnu
Job: WE
Repo location: /scratch1/BMC/zrtrr/rrfs_ci/autoci/pr/1122018556/20221115162014/ufs-srweather-app
Build was Successful
Rocoto jobs started
Long term tracking will be done on 10 experiments
If test failed, please make changes and add the following label back:
ci-hera-gnu-WE
Experiment Succeeded on hera: pregen_grid_orog_sfc_climo
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2
Experiment Succeeded on hera: community_ensemble_2mems_stoch
Experiment Succeeded on hera: grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR
Experiment Succeeded on hera: grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_2017_gfdlmp_regional
Experiment Succeeded on hera: grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_RAP_suite_HRRR
Experiment Succeeded on hera: MET_ensemble_verification
All experiments completed

**Comment by danielabdi-noaa** [comment]
@MichaelLueken I don't think the GNU github actions labels work on either Hera or Jet. We will need to communicate with @venitahagerty  to pass compiler options for `setup_WE2E_tests.sh`.
```
./setup_WE2E_tests.sh machine account compiler test_type
```
In the past it used only the first two, and i don't think it has been updated to use the others.

**Comment by MichaelLueken** [comment]
@danielabdi-noaa Thanks for the heads up!  Looking at the output from the github actions label, I see that it is being built using Intel rather than GNU.

@venitahagerty How difficult is it to update the github actions so that the test will run using:

`./setup_WE2E_tests.sh machine account compiler test_type`

This way, we will be able to test both the Intel and GNU compilers on Hera moving forward.

**Comment by MichaelLueken** [comment]
@natalie-perlin I see that the changes in ufs-weather-model PR [#1468](https://github.com/ufs-community/ufs-weather-model/pull/1468) have been merged.  Would you like for me to update the Hera Intel modulefile to point to the new EPIC-maintained HPC-stack location as part of this PR as well?

**Comment by MichaelLueken** [comment]
On a similar note, @danielabdi-noaa, would you like for me to merge your PR #479 work into this PR?  This way, all modulefile and testing modifications for Hera can be completed with this single PR.

**Comment by danielabdi-noaa** [comment]
@MichaelLueken Sure, please go ahead and merge them.

**Comment by MichaelLueken** [comment]
@natalie-perlin I have updated both the `build_hera_intel.lua` and `build_hera_gnu.lua` to use the EPIC-maintained HPC-stack locations.  The fundament WE2E tests have been run for both GNU and Intel and they both successfully passed. 
 Please let me know if there are any issues that you see with these two files.

@danielabdi-noaa I have merged your work from PR #479 into this PR.  If there are any additional changes that you would like to see, please let me know.

**Review by natalie-perlin — APPROVED** [review]
Thank you for the changes, approving!

**Pull request merged**